File: mobile/blender-buddy/core.bb/Models/CalculationRequest.cs

namespace core.bb.Models
{
    public sealed class CalculationRequest
    {
        public CalculationRequest()
        {
            TopOffGas = new Gas();
            FillSpecs = new TankInfo();
            Residual = new TankInfo();
            TopOffGasType = Models.TopOffGas.Air;
        }

        public MeasureMode System { get; set; }

        public TopOffGas TopOffGasType { get; set; }

        public Gas TopOffGas { get; set; }

        public TankInfo Residual { get; set; }

        public TankInfo FillSpecs { get; set; }
    }
}

File: cordova/Blender-Buddy/scripts/Models/CalculationRequest.ts
import { Gas } from './Gas';
import { TankInfo } from './TankInfo';
import { MeasureMode } from './MeasureMode';
import {TopOffGas } from './TopOffGas';

export class CalculationRequest {
    system: MeasureMode;
    topOffGasType: TopOffGas;
    topOffGas: Gas;
    residual: TankInfo;
    fillSpecs: TankInfo;

    constructor() {
        this.system = MeasureMode.Imperial;
        this.topOffGasType = TopOffGas.Air;
        this.topOffGas = new Gas();
        this.residual = new TankInfo();
        this.fillSpecs = new TankInfo();
    }
}